loader/loader.c

```diff
@@ -767,10 +768,241 @@ VkResult loader_add_to_dev_ext_list(const struct loader_instance *inst, struct l
+    char *env_var_value = loader_secure_getenv(env_var_name, inst);
+    if (NULL != env_var_value) {
+        if (strlen(env_var_value) > 0) {
+            const size_t env_var_len = strlen(env_var_value);
+            // Allocate a separate string since strtok modifies the original string
+            char *parsing_string = loader_instance_heap_calloc(inst, env_var_len + 1, VK_SYSTEM_ALLOCATION_SCOPE_INSTANCE);
+            if (NULL != parsing_string) {
+                const char tokenizer[3] = ",";
+
+                for (uint32_t iii = 0; iii < env_var_len; ++iii) {
+                    parsing_string[iii] = (char)tolower(env_var_value[iii]);
+                }
+                parsing_string[env_var_len] = '\0';
+
+                char *token = strtok(parsing_string, tokenizer);
+                while (NULL != token) {
+                    enum loader_filter_string_type cur_filter_type;
+                    const char *actual_start;
+                    size_t actual_len;
+                    determine_filter_type(token, &cur_filter_type, &actual_start, &actual_len);
+                    if (actual_len > VK_MAX_EXTENSION_NAME_SIZE) {
+                        strncpy(filter_struct->filters[filter_struct->count].value, actual_start, VK_MAX_EXTENSION_NAME_SIZE);
+                    } else {
+                        strncpy(filter_struct->filters[filter_struct->count].value, actual_start, actual_len);
+                    }
+                    filter_struct->filters[filter_struct->count].length = actual_len;
+                    filter_struct->filters[filter_struct->count++].type = cur_filter_type;
+                    if (filter_struct->count >= MAX_ADDITIONAL_FILTERS) {
+                        break;
+                    }
+                    token = strtok(NULL, tokenizer);
+                }
+                loader_instance_heap_free(inst, parsing_string);
+            } else {
+                loader_log(inst, VULKAN_LOADER_ERROR_BIT, 0,
+                           "parse_generic_filter_environment_var: Failed to allocate space for parsing env var \'%s\'",
+                           env_var_name);
+            }
+        }
+        loader_free_getenv(env_var_value, inst);
+    }
+}
+
+// Parse the disable layer string.  The layer disable has some special behavior because we allow it to disable
+// all layers (either with "~all~", "*", or "**"), all implicit layers (with "~implicit~"), and all explicit layers
+// (with "~explicit~"), in addition to the other layer filtering behavior.
+void parse_layers_disable_filter_environment_var(const struct loader_instance *inst,
+                                                 struct loader_envvar_disable_layers_filter *disable_struct) {
+    memset(disable_struct, 0, sizeof(struct loader_envvar_disable_layers_filter));
+    char *env_var_value = loader_secure_getenv(VK_LAYERS_DISABLE_ENV_VAR, inst);
+    if (NULL != env_var_value) {
+        if (strlen(env_var_value) > 0) {
```

Comment:
Same as above, this should early-out in the error cases.

Reply:
👍🏽